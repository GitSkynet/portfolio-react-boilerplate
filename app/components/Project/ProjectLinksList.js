import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import ProjectExternalLink from './ProjectExternalLink';

const StyledList = styled.ul`
  display: flex;
  flex-flow: column nowrap;
  padding: 1rem;
  list-style: none;
  text-align: center;

  @media (min-width: 660px) {
    display: flex;
    flex-flow: row nowrap;
    justify-content: center;
  }
`;

const Wrapper = styled.div`
  left: 50%;
  margin: 0 !important;
  overflow: hidden;
  position: absolute;
  top: 50%;
  transform: translate3d(-50%, -50%, 0);
  width: 100%;
  z-index: 1;
  backdrop-filter: ${(props) => props.hasFocus ? 'blur(5px)' : 'unset'};
  background-color: rgba(255, 255, 255, 0.5);
  opacity: ${(props) => props.hasFocus ? 1 : 0};
  text-align: center;
  transition: opacity 0.3s 0s ease-in-out;
`;

function ProjectLinksList(props) {
  return (
    <Wrapper hasFocus={props.hasFocus}>
      <StyledList>
        {
          props.links.map(
            ({ name, url /* , faIcon */ }) => (
              <ProjectExternalLink
                key={name}
                name={name}
                url={url}
                hasFocus={props.hasFocus}
              />
            )
          )
        }
      </StyledList>
    </Wrapper>
  );
}

ProjectLinksList.propTypes = {
  links: PropTypes.array.isRequired,
  hasFocus: PropTypes.bool,
};

ProjectLinksList.defaultProps = {
  hasFocus: false,
};

export default ProjectLinksList;
