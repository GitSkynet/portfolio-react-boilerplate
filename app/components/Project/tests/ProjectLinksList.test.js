import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';

import ProjectLinksList from '../ProjectLinksList';

describe('ProjectLinksList', () => {
  const links = [
    {
      name: 'github',
      url: 'https://github.com/',
      faIcon: 'github',
    },
    {
      name: 'codepen',
      url: 'https://codepen.io/',
      faIcon: 'codepen',
    },
  ];

  let wrapper;

  beforeEach(() => {
    wrapper = toJson(shallow(<ProjectLinksList links={links} />));
  });

  it('should render and match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
