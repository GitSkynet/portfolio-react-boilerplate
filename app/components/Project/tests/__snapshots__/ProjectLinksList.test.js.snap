// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectLinksList should render and match the snapshot 1`] = `
<ProjectLinksList__Wrapper
  hasFocus={false}
>
  <ProjectLinksList__StyledList>
    <ProjectExternalLink
      faIcon="external-link"
      hasFocus={false}
      key="github"
      name="github"
      url="https://github.com/"
    />
    <ProjectExternalLink
      faIcon="external-link"
      hasFocus={false}
      key="codepen"
      name="codepen"
      url="https://codepen.io/"
    />
  </ProjectLinksList__StyledList>
</ProjectLinksList__Wrapper>
`;
